data/src/main/java/com/example/bestapplication/Mapper.kt

```diff
@@ -0,0 +1,113 @@
+package com.example.bestapplication
+
+import com.example.bestapplication.model.ActorApi
+import com.example.bestapplication.model.GenreApi
+import com.example.bestapplication.model.MovieFullApi
+import com.example.bestapplication.model.MoviePreviewApi
+import com.example.bestapplication.database.MovieEntity
+import com.example.bestapplication.favorite_movie.entity.MovieDatabaseEntity
+import com.example.bestapplication.movie_details.entity.Actor
+import com.example.bestapplication.movie_details.entity.MovieFull
+import com.example.bestapplication.movie_list.entity.Genre
+import com.example.bestapplication.movie_list.entity.MoviePreview
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+
+object Mapper {
+
+
+    @ExperimentalSerializationApi
+    fun mapMoviesListToDb(movie: MovieFullApi): MovieEntity {
+
+        val genres = Json.encodeToString(movie.genres)
+        return MovieEntity(
+            movie.id,
+            movie.poster,
+            movie.title,
+            movie.ratings,
+            genres,
+            movie.runtime
+        )
+    }
+
+    fun mapMoviesInDb(movie: MovieDatabaseEntity): MovieEntity {
+
+        return MovieEntity(
+            movie.id,
+            movie.poster,
+            movie.title,
+            movie.ratings,
+            movie.genres,
+            movie.runtime
+        )
+    }
+
+    fun mapMoviesFromDb(movie: List<MovieEntity>): List<MovieDatabaseEntity> {
```

Comment:
Should be like param type MovieEntity and return type MovieDatabaseEntity and then invoke like someList.map{ mapMovieEntityToDb(it) }